sdk/deepbook-v3/src/client.ts

```diff
@@ -92,51 +86,64 @@ export class DeepBookClient {
 	};
 
 	/// Balance Manager
-	createAndShareBalanceManager() {
-		return this.#signAndExecuteCommand(this.#config.balanceManager.createAndShareBalanceManager());
+	createAndShareBalanceManager(tx: Transaction = new Transaction()) {
+		return this.#config.balanceManager.createAndShareBalanceManager(tx);
 	}
 
-	depositIntoManager(managerKey: string, amountToDeposit: number, coinKey: CoinKey) {
+	depositIntoManager(
+		managerKey: string,
+		amountToDeposit: number,
+		coinKey: string,
+		tx: Transaction = new Transaction(),
+	) {
 		const balanceManager = this.#getBalanceManager(managerKey);
 		const coin = this.#config.getCoin(coinKey);
 
-		return this.#signAndExecuteCommand(
-			this.#config.balanceManager.depositIntoManager(balanceManager.address, amountToDeposit, coin),
+		return this.#config.balanceManager.depositIntoManager(
+			balanceManager.address,
+			amountToDeposit,
+			coin,
+			tx,
 		);
 	}
 
-	withdrawFromManager(managerKey: string, amountToWithdraw: number, coinKey: CoinKey) {
+	withdrawFromManager(
+		managerKey: string,
+		amountToWithdraw: number,
+		coinKey: string,
+		tx: Transaction = new Transaction(),
+	) {
 		const balanceManager = this.#getBalanceManager(managerKey);
 		const coin = this.#config.getCoin(coinKey);
 
 		const recipient = this.getActiveAddress();
 
-		return this.#signAndExecuteCommand(
-			this.#config.balanceManager.withdrawFromManager(
-				balanceManager.address,
-				amountToWithdraw,
-				coin,
-				recipient,
-			),
+		return this.#config.balanceManager.withdrawFromManager(
+			balanceManager.address,
+			amountToWithdraw,
+			coin,
+			recipient,
+			tx,
 		);
 	}
 
-	withdrawAllFromManager(managerKey: string, coinKey: CoinKey) {
+	withdrawAllFromManager(managerKey: string, coinKey: string, tx: Transaction = new Transaction()) {
 		const balanceManager = this.#getBalanceManager(managerKey);
 		const coin = this.#config.getCoin(coinKey);
 
 		const recipient = this.getActiveAddress();
-		return this.#signAndExecuteCommand(
-			this.#balanceManager.withdrawAllFromManager(balanceManager.address, coin, recipient),
-		);
+		return this.#balanceManager.withdrawAllFromManager(balanceManager.address, coin, recipient, tx);
 	}
 
-	async checkManagerBalance(managerKey: string, coinKey: CoinKey) {
+	async checkManagerBalance(
+		managerKey: string,
+		coinKey: string,
+		tx: Transaction = new Transaction(),
```

Comment:
passing in a transaction with existing commands will mean the result indexes are wrong. and we won't be able to parse the balance